components/ei-analytics/org.wso2.analytics.solutions.ei.analytics/widgets/EIAnalyticsSearchBox/src/EIAnalyticsSearchBox.jsx

```diff
@@ -141,39 +314,173 @@ class EIAnalyticsSearchBox extends Widget {
+        super.getWidgetConfiguration(this.props.widgetID)
+            .then((message) => {
+                //based on the component type, query ESB or Mediator stat tables
+                if (this.pageName == this.pgAPI || this.pageName == this.pgProxy || this.pageName == this.pgInbound) {
+                    query = message.data.configs.providerConfig.configs.config.queryData.queryESB;
+
+                    //change pageName variable to 'Proxy Service' to query data based on the componentType
+                    if (this.pageName == this.pgProxy) {
+                        componentType = 'proxy service';
+                    }
+                    //change pageName variable to 'Inbound EndPoint'to query data based on the componentType
+                    else if (this.pageName == this.pgInbound) {
+                        componentType = 'inbound endpoint';
+                    }
+
+                } else {
+                    query = message.data.configs.providerConfig.configs.config.queryData.queryMediator;
+                }
+                message.data.configs.providerConfig.configs.config.queryData.query = query
+                    .replace('{{paramComponentType}}', componentType);
+                super.getWidgetChannelManager().subscribeWidget(this.props.id,
+                    this.handleDataReceived, message.data.configs.providerConfig);
+
+            })
+            .catch((error) => {
+                this.setState({
+                    faultyProviderConf: true
+                });
+            });
+    }
+
+    componentWillUnmount() {
+        super.getWidgetChannelManager().unsubscribeWidget(this.props.id);
     }
 
     render() {
+        const { classes } = this.props;
+        this.updateTextBoxColor();
         return (
-            <div>
-                <Select
-                    name="form-field-name"
-                    onChange={this.handleChange}
-                    options={this.state.optionArray}
-                    placeholder={this.state.selectedOption}
-                    value={this.state.selectedOption}
-                    clearable={false}
-                >
-                </Select>
-            </div>
+            <JssProvider
+                generateClassName={generateClassName}>
+                <MuiThemeProvider
+                    theme={this.props.muiTheme.name === 'dark' ? darkTheme : lightTheme}>
+                    <Scrollbars
+                        style={{height: this.state.height}}>
+                        <div
+                            style={{
+                                paddingLeft: 15,
+                                paddingRight: 15,
+                                paddingTop: 5,
+                            }}>
+                            <div>
+                                Filter by {this.formatPageName(this.pageName)}
+                            </div>
+                            <div>
+                                <Select
```

Comment:
@ruwiniwj Please add a TODO to move this `<Select>` component and related stuff to a separate component as this class have become very long.

Reply:
Fixed with https://github.com/wso2/analytics-solutions/pull/114